synapse/http/client.py

```
@@ -145,6 +145,9 @@ def post_json_get_json(self, uri, post_json):
 
         body = yield preserve_context_over_fn(readBody, response)
 
+        if response.code / 100 != 2:
+            raise CodeMessageException(response.code, body)
```

do we really want to raise an exception on 300s?